### ISSUE TITLE
Exporting and importing stages trees

### ISSUE THREAD

**Issue by felixleopoldo** [stagedtrees/stagedtrees#132]
Hi,

Thanks for providing this package. I wonder if there is a way to import/export staged trees, i.e. is it possible to export to a file that can then be imported?

Best

Felix

**Comment by gherardovarando** [comment]
Hi,

you can save and read an Rdata file. 

If you want to save an object of class sevt to import it from another programming language, we do not have currently a method to export sevt objects.
Nevertheless, I tried and you can save it to JSON with (example with the Titanic data)
```
model <- full(Titanic)
cat(rjson::toJSON(model), file = "filename.json")
``` 

this works for the stages structures and partially for the estimated probabilities. 


you can then read them back as a sevt object with
```
model2 <- rjson::fromJSON(readLines(file("filename.json")), warn = FALSE)
class(model2) <- "sevt"
```
but the `prob` and `ctables` fields (where are stored the estimated probabilities and the data) are messed up since they are more complex R objects then simple list and thus JSON can't represent them properly. 

anyway you can fix them by refitting the model with

```
model2 <- sevt_fit(model2, Titanic)
``

**Comment by felixleopoldo** [comment]
Ok, thanks! If I would like to export the full probability distribution (maybe after taking logarithms) to a CSV file, can that be done in some neat way?

An example of 2 binary variables, X1 and X2 could be something like:

X1,X2,prob
0,0,0.2
0,1,0.3
1,0,0.1
1,1,0.4

**Comment by gherardovarando** [comment]
you are welcome, 

yes sure this is very easy to do 
here is the code (the second option for generating the sample space in `space` is to order the space with the classical lexicographic ordering):

```
## generate the full sample space
space <- expand.grid(model$tree)
## or with differetn ordering
# space <- rev(expand.grid(rev(model$tree)))

prob <- prob(model, space)
write.csv(cbind(space, prob = prob), file = "probabilities.csv", row.names = FALSE)
```

if you want logarithms then just  replace the probability computations with 
```
prob <- prob(mode, space, log = TRUE)
```

**Comment by gherardovarando** [comment]
Thanks a lot for the suggestion.

It would be interesting, but how do you deal with different cardinalities of the variables ? 
Do you add NaNs at the end of the data frame ?

**Comment by felixleopoldo** [comment]
Yes exactly, e.g. the PROB_2 column would be NaN for binary variables but it contains a probability for ternary variables if such exists.
